src/pages/library/datasetBuilder/DatasetBuilder.ts

```diff
@@ -0,0 +1,378 @@
+import { useLoadedData } from 'src/libs/ajax/loaded-data/useLoadedData';
+import colors from 'src/libs/colors';
+import * as Nav from 'src/libs/nav';
+import { datasetBuilderCohorts, datasetBuilderConceptSets } from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+import { StringInput } from 'src/pages/library/data-catalog/CreateDataset/CreateDatasetInputs';
+import {
+  PAGE_PADDING_HEIGHT,
+  PAGE_PADDING_WIDTH,
+  PREPACKAGED_CONCEPT_SETS,
+} from 'src/pages/library/datasetBuilder/constants';
+import { Cohort, ConceptSet, DatasetBuilderType } from 'src/pages/library/datasetBuilder/dataset-builder-types';
+import { DatasetBuilderHeader } from 'src/pages/library/datasetBuilder/DatasetBuilderHeader';
+import { validate } from 'validate.js';
+
+const DatasetBuilderSelectorSubHeader = ({ children }) => div({ style: { fontSize: 12, fontWeight: 600 } }, children);
+
+interface ValuesSet<T extends DatasetBuilderType> {
+  header: string;
+  values: T[];
+}
+
+interface DatasetBuilderSelectorProps<T extends DatasetBuilderType> {
+  number: number;
+  header: string;
+  subheader?: string;
+  valuesSets: ValuesSet<T>[];
+  selectedValuesSets: ValuesSet<T>[];
+  onChange: (newValuesSets: ValuesSet<T>[]) => void;
+  headerAction: any;
+  placeholder?: any;
+  width?: string | number;
+  maxWidth?: number;
+}
+const DatasetBuilderSelector = <T extends DatasetBuilderType>({
+  number,
+  header,
+  subheader,
+  headerAction,
+  placeholder,
+  valuesSets,
+  onChange,
+  selectedValuesSets,
+  width = '30%',
+}: DatasetBuilderSelectorProps<T>) => {
+  return div({ style: { width, marginTop: '1rem' } }, [
+    div({ style: { display: 'flex', width: '100%', justifyContent: 'space-between', alignItems: 'flex-start' } }, [
+      div({ style: { display: 'flex' } }, [
+        div(
+          {
+            style: {
+              backgroundColor: colors.accent(0.2),
+              color: colors.dark(),
+              padding: '0.5rem',
+              borderRadius: '2rem',
+              fontSize: 14,
+              height: 24,
+              width: 24,
+              display: 'flex',
+              alignItems: 'center',
+              justifyContent: 'center',
+              fontWeight: 600,
+            },
+          },
+          [number]
+        ),
+        div({ style: { marginLeft: 10, height: '3rem' } }, [
+          h3({ style: { marginTop: 0, marginBottom: '0.5rem' } }, [header]),
+          div({ style: { fontSize: 12 } }, [subheader]),
+        ]),
+      ]),
+      headerAction,
+    ]),
+    div(
+      {
+        style: {
+          backgroundColor: 'white',
+          border: `1px solid ${colors.dark(0.5)}`,
+          borderRadius: 10,
+          height: 300,
+          padding: '1rem',
+          marginTop: '0.5rem',
+        },
+      },
+      [
+        valuesSets && valuesSets.length > 0 && _.flatMap((valuesSet) => valuesSet.values, valuesSets).length > 0
```

Comment:
Do you need to check `valuesSets.length > 0` before calling `flatMap`? It seems like this would be equivalent:
```suggestion
        valuesSets && _.flatMap((valuesSet) => valuesSet.values, valuesSets).length > 0
```